main/src/main.ts

```diff
@@ -103,8 +103,27 @@ export async function blockQuit(source: string, event?: Electron.Event) {
 
   // Only send graceful-exit message if mainWindow is still valid
   try {
-    if (mainWindow && !mainWindow.isDestroyed()) {
+    if (!mainWindow || mainWindow.isDestroyed()) {
+      log.info('MainWindow destroyed, recreating for graceful shutdown...')
+      mainWindow = createWindow()
+    }
+
+    if (mainWindow) {
+      log.info('Showing window for graceful shutdown...')
+
+      if (mainWindow.isMinimized()) {
+        mainWindow.restore()
+      }
+
+      mainWindow.show()
+      mainWindow.focus()
+
+      await pollWindowReady(mainWindow)
+
       mainWindow.webContents.send('graceful-exit')
+
+      // Give renderer time to navigate to shutdown page
+      await new Promise((resolve) => setTimeout(resolve, 500))
```

Comment:
I think for this type of thing we should introduce an "async sleep function" that can be called like this 


```ts
await sleep(500)
```


basically the code is the same, I would just it in some utils file and make it generic


```ts
export async function sleep(delayMiliseconds: number) {
  return new Promise(resolve => setTimeout(resolve, delayMiliseconds)
}
```

Reply:
makes sense